src/QLContext.actor.cpp

```diff
@@ -185,22 +185,20 @@ struct FDBPlugin : ITDoc, ReferenceCounted<FDBPlugin>, FastAllocated<FDBPlugin>
 	void addref() override { ReferenceCounted<FDBPlugin>::addref(); }
 	void delref() override { ReferenceCounted<FDBPlugin>::delref(); }
 
-	std::pair<bool, Reference<DocumentDeferred>> findOrCreate(Reference<DocTransaction> tr, DataKey const& key) {
-		if (key.size() > 1) {
-			std::string documentPrefix = key.keyPrefix(2).toString();
-			auto info = tr->infos.find(documentPrefix);
-			if (info == tr->infos.end())
-				info = tr->infos
-				           .insert(std::make_pair(documentPrefix, Reference<DocumentDeferred>(new DocumentDeferred())))
-				           .first;
-			return std::make_pair(true, (*info).second);
-		}
-		return std::make_pair(false, Reference<DocumentDeferred>());
+	Reference<DocumentDeferred> findOrCreate(Reference<DocTransaction> tr, DataKey const& key) {
+		ASSERT(key.size() > 1);
+		std::string documentPrefix = key.keyPrefix(2).toString();
+		auto info = tr->infos.find(documentPrefix);
```

Comment:
I'd suggest also rename the `info` to something that makes sense, like `docMutationBuffers`. This is non blocking though.